src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -448,20 +475,29 @@
         this)
       (a/stop [this]
         (shutdown! app-context)
-        this))))
+        this)
+      (a/restart [this]
+        (try
+          (run-lifecycle-fns app-context s/stop "stop" (reverse ordered-services))
```

Comment:
> @camlow325 this implementation actually does perform all stops, even if one errors.  

`run-lifecycle-fns` will catch an exception but then will re-throw it, causing the rest of the `stop` calls to not be made in this case:

https://github.com/puppetlabs/trapperkeeper/blob/1.1.3/src/puppetlabs/trapperkeeper/internal.clj#L186

So I think the first time an exception is encountered, control will fall through to catch block, where the `shutdown-promise` is delivered.

Although, as I mentioned in a separate comment, I think `stop` will end up getting called a second time as part of the shutdown logic on the Trapperkeeper main thread after the shutdown promise is handled.  So maybe my comment here is moot.  I'm actually a little more concerned about the double-stop call case.  Essentially, I think it will mean that all service implementation `stop` methods need to be resilient to being called multiple times without an init/start in between.  Maybe that's not the end of the world either, though.  Still curious what @cprice404 thinks about this.